### PR TITLE
Added explicit conversion APIs for Span to ROSpan and Memory to ROMemory

### LINKED ISSUE #24564: Add explicit conversion APIs for Span to ROSpan and Memory to ROMemory
## Rationale
From https://github.com/dotnet/corefx/issues/23862#issuecomment-328097809

```C#
public static implicit operator ReadOnlyMemory<T>(Memory<T> memory);
public static implicit operator ReadOnlySpan<T> (Span<T> span);
```

This is the only way to convert from `Span<T>` to `ReadOnlySpan<T> and `Memory<T>` to `ReadOnlyMemory<T>`. [Framework Design Guildelines say](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/operator-overloads):

> ✓ CONSIDER providing methods with friendly names that correspond to each overloaded operator.

Specifically, F# does not support implicit operators, but allows calling them as `op_Implicit`. The only way to convert from `Memory<T>` to `ReadOnlyMemory<T>` in F# would be the following?

```F#
let memoryToReadOnlyMemory (memory : Memory<'a>) : ReadOnlyMemory<'a> =
    Memory<'a>.op_Implicit memory
```

## Proposed API
Therefore, add these explicit conversion APIs to `Span<T>` and `Memory<T>`:
```C#
public static ReadOnlyMemory<T> AsReadOnlyMemory(this Memory<T> memory);
public static ReadOnlySpan<T> AsReadOnlySpan(this Span<T> span);
```

cc @svick, @KrzysztofCwalina, @jkotas, @terrajobst

### PR DESCRIPTION
**Pull request by zaytsev-victor**
Added explicit conversion APIs to ```Span<T>``` and ```Memory<T>```:
```csharp
public static ReadOnlyMemory<T> AsReadOnlyMemory(this Memory<T> memory);
public static ReadOnlySpan<T> AsReadOnlySpan(this Span<T> span);
```
Resolves #24564
I'm not sure if tests required.

### CODE REVIEW

File: src/System.Memory/src/System/MemoryExtensions.cs [hunk 1]
```diff
@@ -249,6 +249,12 @@ public static ReadOnlySpan<T> AsReadOnlySpan<T>(this T[] array)
             return new ReadOnlySpan<T>(array);
         }
 
+        /// <summary>
+        /// Creates a new readonly span over the entire target span.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```
Comment (stephentoub) [inline src/System.Memory/src/System/MemoryExtensions.cs]: This shouldn't be necessary.  Same below.

**Comment by stephentoub** [comment]
> I'm not sure if tests required.

Please and thank you.

File: src/System.Memory/tests/Memory/AsReadOnlyMemory.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.MemoryTests
+{
+    public static partial class AsReadOnlyMemory
```
Comment (ahsonkhan) [inline src/System.Memory/tests/Memory/AsReadOnlyMemory.cs]: Does this need to be a partial class?

File: src/System.Memory/tests/System.Memory.Tests.csproj [hunk 3]
```diff
@@ -9,6 +9,7 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Release|AnyCPU'" />
   <ItemGroup>
     <Compile Include="AllocationHelper.cs" />
+    <Compile Include="Memory\AsReadOnlyMemory.cs" />
```
Comment (ahsonkhan) [inline src/System.Memory/tests/System.Memory.Tests.csproj]: nit: Can you move this down to the ItemGroup with the rest of the Memory tests? i.e. here: https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/System.Memory.Tests.csproj#L80

**Pull request merged**